sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/PipelinedQueryExecutionContextBase.java

```diff
@@ -127,7 +146,52 @@ protected static <T> BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<ID
         }
     }
 
+    protected static <T> BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createCommonHybridPipelineComponentFunction(
+        BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createBaseComponent,
+        HybridSearchQueryInfo hybridSearchQueryInfo) {
+
+        BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createSkipComponentFunction;
+        if (hybridSearchQueryInfo.hasSkip()) {
+            createSkipComponentFunction =
+                (continuationToken, documentQueryParams) ->
+                    SkipDocumentQueryExecutionContext.createAsync(createBaseComponent,
+                        hybridSearchQueryInfo.getSkip(),
+                        continuationToken,
+                        documentQueryParams);
+        } else {
+            createSkipComponentFunction = createBaseComponent;
+        }
+
+        BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createLimitComponentFunction;
+        if (hybridSearchQueryInfo.hasTake() && hybridSearchQueryInfo.hasSkip()) {
+            createLimitComponentFunction =
+                (continuationToken, documentQueryParams) ->
+                    TakeDocumentQueryExecutionContext.createAsync(createSkipComponentFunction,
+                        hybridSearchQueryInfo.getTake(),
+                        continuationToken,
+                        documentQueryParams,
+                        TakeDocumentQueryExecutionContext.TakeEnum.LIMIT);
+        } else {
+            createLimitComponentFunction = createSkipComponentFunction;
```

Comment:
what if it has take but not skip?

Reply:
If there is only take, that being checked in line 178, and the logic of it's implementation is in lines 179-184. And I have added an additional check in DocumentQueryExecutionFactory that if skip is there and take is not there to throw an exception.